service/history/queues/executable.go

```diff
@@ -209,7 +208,7 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 			defer e.Unlock()
 
 			e.attempt++
-			if e.attempt > e.criticalRetryAttempt() {
+			if e.attempt > taskCriticalAttempts {
```

Comment:
Log message is bad. "Critical" is already in the tag and number of attempts is missing (but might be useful):
```
e.logger.Error("Error processing task after number of attempts.", tag.Error(err), tag.Attempt(e.attempt), tag.OperationCritical)
```

Reply:
I am not sure how useful # of attempts is in the log given that we have the attempt count metric. But I can add it.
 
Re. the message itself, I kinda agree. My main concern is oncall needs to memorize a new way for finding this important log (this message is used/searched a lot). So if it's not something you have strong opinion on, I'd like to leave it as is.